src/js/core/factories/Grid.js

```diff
@@ -1891,7 +1891,7 @@ angular.module('ui.grid')
       if (typeof(row.entity['$$' + col.uid]) !== 'undefined') {
         col.cellDisplayGetterCache = $parse(row.entity['$$' + col.uid].rendered + custom_filter);
       } else if (this.options.flatEntityAccess && typeof(col.field) !== 'undefined') {
-        col.cellDisplayGetterCache = $parse(row.entity[col.field] + custom_filter);
+        col.cellDisplayGetterCache = $parse("'" + row.entity[col.field] + "'" + custom_filter);
```

Comment:
For my example where it doesn't work, `row.entity[col.field]` is already a string. 

It's just that `$parse` throws a Syntax Error: Token is unexpected, and wrapping the expression in quotes fixes this.

Although thinking about it now, this still doesn't work if `row.entity[col.field]` contains a single quote. So I guess this needs to be escaped.